text/trace/0000-error_flagging.md

```diff
@@ -0,0 +1,86 @@
+# Error Flagging with Status Codes
+This proposal adds two status codes explicitly for use as overrides by the end user, and proposes a canonical mapping of semantic conventions to status codes. This clarifies how error reporting should work in OpenTelemetry.
+ 
+## Motivation
+Error reporting is a fundamental use case for distributed tracing. While we prefer that error flagging occurs within analysis tools, and not within instrumentation plugins, a number of currently supported analysis tools and protocols rely on the existence of an explicit error flag reported from instrumentation. In OpenTelmetry, the error flag is called "status codes."
+ 
+However, there is confusion over the mapping of semantic conventions to status codes, and concern over the subjective nature of errors. Which network failures count as an error? Are 404s an error? The answer is dependent on the situation.
+ 
+There is one major exception. Both application developers and operators have a deep understanding of what constitutes an error in their system. OpenTelemetry must provide a way for these users to control error flagging, and explicitly indicate both when a span should and should not be counted as an error.
+A second exception is supporting analysis tools which require explicit error flagging in the data which they receive. In this case, an operator must be able to apply an error flagging schema at some point during the OTLP data processing pipeline.
+ 
+## Explanation
+The following changes add several missing features required for proper error reporting, and are completely backwards compatible with OpenTelemetry today.
+ 
+### Status Codes
+The following status codes are added to our current  schema.
+ 
+* `DEFAULT` No status has been set. Any errors must be detected by the analysis tool. (This replaces `OK`.)
+* `ERROR` Instrumentation has marked a span as an error. (This replaces `UNKNOWN`.)
```

Comment:
How does the instrumentation make this determination? It appears this is left to discretion but clearly needs to be consistent across client libraries. For example, should `http.status_code` in the range 500-599 automatically be marked as an error?

Reply:
Note that ERROR != ERROR_OVERRIDE. I think this leaves the status quo of having  http.status_code in the range 500-599 being mapped to various status codes (none of which is UNKNOWN/ERROR): https://github.com/open-telemetry/opentelemetry-specification/blob/821118e5/specification/trace/semantic_conventions/http.md#status

Reply:
@flands the "Error Mapping" section tries to address this. We should provide a schema that explicitly describes how every semantic convention affects the status code. Basically, take the HTTP mapping example above, and expand it into a new page in the spec which contains all of the mappings.